Assets/Scripts/Interactables/Dialogue/DialogueInteractable.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DialogueInteractable : Interactable
+{
+    public Conversation conversation;
+
+    private GameObject NPCBubble;
+    private GameObject BirdieBubble;
+    private SpeakerUI speakerUILeft;
+    private SpeakerUI speakerUIRight;
+
+    private int activeLineIndex = 0;
+
+    public override void OnInteract(GameObject birdie)
+    {
+        BirdieBubble = birdie.transform.GetChild(1).gameObject; 
+        NPCBubble = this.transform.GetChild(1).gameObject;
+
+        speakerUILeft = BirdieBubble.GetComponent<SpeakerUI>();
+        speakerUIRight = NPCBubble.GetComponent<SpeakerUI>();
+
+        HideInteractUI();
+        AdvanceConversation();
+
+        base.OnInteract(birdie);
+    }
+
+    void AdvanceConversation() {
+        if (activeLineIndex < conversation.lines.Length)
+        {
+            //If there is still conversation left
+            DisplayLine();
+            activeLineIndex += 1;
+        }
+        else
+        {
+            //Once the conversation is over 
+            //what happens 
+            speakerUILeft.Hide();
+            speakerUIRight.Hide();
+            activeLineIndex = 0;
+        }
+    }
+
+    void DisplayLine() {
+        Line line = conversation.lines[activeLineIndex];
+        bool isBirdie = line.isBirdie;
+
+        if (isBirdie)
+        {
+            SetDialogue(speakerUILeft, speakerUIRight, line.text);

```

Comment:
Looking at this logic, it's gonna be confusing to remember, left = birdie, right = not birdie, so can we rename this variable to `speakerUIBirdie` and `speakerUINPC` or something similar?

Reply:
Yes!! I should do that :)